### PR TITLE
Update CSV, DAT, SQL output module to current data model

### LINKED ISSUE #101: [BUG] SQL output module is not up to date with current internal data model
**Describe the bug**
The SQL output module `tops_sql` is severely out of date with respect to our current data model.

**To Reproduce**
Try exporting from any input format to SQL. The program fails with

```
TypeError: cannot unpack non-iterable Feature object
```

**Expected behavior**
A well-formed SQL file is created.

**Desktop (please complete the following information):**
 - Operating system: both Linux and Windows
 - Software version: 0.5.0dev

### LINKED ISSUE #103: [BUG] DAT output module is out of date with current data model
**Describe the bug**
The DAT output module `tops_dat` is severely out of date with respect to our current data model.

**To Reproduce**
Try exporting from any input format to DAT. The program fails with

```
TypeError: 'Feature' object is not subscriptable
```

**Expected behavior**
A well-formed DAT file is created.

**Desktop (please complete the following information):**
 - Operating system: both Linux and Windows
 - Software version: 0.5.0dev

### LINKED ISSUE #104: [BUG] DAT output module has undocumented filter on processed data
**Describe the bug**
The DAT output module `tops_dat` adds an undocumented filter on input data points, keeping only those that have a `desc` string ending with `R`. This is directly related to the specific use of the output format that was done in the GQB archaeological excavations, for integration with the Archis photorectification software.

**To Reproduce**
1. Prepare an input file with any number of points
2. Convert to DAT
3. Only points with description ending in "R" are converted (if any)
4. If no points satisfying the above condition are found, an empty file is created

**Expected behavior**
All points should be processed and included in the output DAT file.

**Desktop (please complete the following information):**
 - Operating system: Linux
 - Software version 0.5.0dev

**Additional context**

### LINKED ISSUE #105: [BUG] CSV output module expects "point_name" property that is often missing
**Describe the bug**
CSV output module expects "point_name" property that is often missing, because only a few input formats actually return features with that property.

**To Reproduce**
1. Prepare input data in any format of Leica TCR, Sokkia SDR, Topcon GTS, Trimble, Zeiss
2. Convert to CSV
3. Program fails with `KeyError: 'point_name'`

**Expected behavior**
`point_name` should be processed only when present in input data.

**Desktop (please complete the following information):**
 - Operating system: Linux
 - Software version 0.5.0dev

**Additional context**
Only Leica GSI, Carlson RW5 and Nikon RAW formats have a documented "point name" feature. It's not clear what the difference is between "point name" and "desc" (standard short variable name for "description"). "desc" is implemented in all formats, but for these three formats it is more similar to a "category label" (e.g. not unique or sequential or manually entered text).

### PR DESCRIPTION
**Pull request by steko**
- Fixes #101 
- Fixes #103
- Fixes #104
- Fixes #105

### CODE REVIEW

**Comment by steko** [comment]
Uhm, perhaps it's easier to fix this entire set of related bugs in one pull request...

**Review by psolyca — CHANGES_REQUESTED** [review]
Maybe you could change all Copyright year to be homogeneous.

File: totalopenstation/output/tops_sql.py [hunk 1]
```diff
@@ -29,16 +29,17 @@ def to_sql(point, tablename):
...
     sql_string = "INSERT INTO %(tablename)s" % params
     sql_string += "(point_id, point_geom, point_text) VALUES"
     sql_string += "(%(pid)s,GeomFromText('%(wkt)s'),'%(text)s');\n" % params
     return sql_string
 
 
 def to_wkt(point):
-    pid, x, y, z, text = point
+    x = point.geometry.x
+    y = point.geometry.y
```
[the remark this answers is not in the document]
Comment (steko) [reply totalopenstation/output/tops_sql.py:42]: I'm finding some documentation where the syntax is described as `POINT Z (1.0 2.0 3.0)`. Would you like this fix included in this PR @psolyca?

[the remark this answers is not in the document]
Comment (psolyca) [reply totalopenstation/output/tops_sql.py:42]: Yes, we could add it here.

**Pull request merged**